### PR TITLE
Allow block restrictions for Thermochimica calculations

### PR DESCRIPTION
**Pull request by parikshitbajpai**
## Reason
We might want to run Thermochimica on specific blocks in the mesh but currently the ChemicalCompositionAction automatically sets-up required variables & UOs on all blocks of the mesh. 

## Design
Add `BlockRestrictable::validParams` to `ChemicalCompositionAction::validParams` enabling us to specify `block=<block_id>`.

## Impact
Supports more use cases.

### CODE REVIEW

**Comment by parikshitbajpai** [comment]
@dschwen @mtnwalker

File: modules/chemical_reactions/test/tests/thermochimica/tests [hunk 1]
```diff
@@ -47,6 +47,16 @@
     required_submodule = 'contrib/thermochimica'
   []
 
+  [MoRu_block-restrict]
+    type = Exodiff
+    input = MoRu_block-restrict.i
+    exodiff = MoRu_block-restrict_out.e
+    design = 'ChemicalCompositionAction.md '
```
Comment (GiudGiud) [inline modules/chemical_reactions/test/tests/thermochimica/tests:54]: missing the issues field

File: modules/chemical_reactions/test/tests/thermochimica/tests [hunk 2]
```diff
@@ -47,6 +47,16 @@
...
+    requirement = 'The system shall be able to run Thermochimica on specific blocks'
```
Comment (GiudGiud) [inline modules/chemical_reactions/test/tests/thermochimica/tests:55]:
```suggestion
    requirement = 'The system shall be able to only run Thermochimica on specific blocks'
```

**Review by dschwen — APPROVED** [review]
Looks good to me!

**Comment by GiudGiud** [comment]
You're assigned @dschwen btw

**Pull request merged**